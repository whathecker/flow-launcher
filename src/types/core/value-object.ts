export type PriorityTier = "n/a" | "highest" | "high" | "mid" | "low";

export type PriorityMeasure = "n/a" | "yes" | "no"; // TODO: consider to use MIXED type (boolean and null) instead?

export class Priority {
  constructor(
    public tier: PriorityTier,
    public importance: PriorityMeasure,
    public urgency: PriorityMeasure,
  ) {}
}
