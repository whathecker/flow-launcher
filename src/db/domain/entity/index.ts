import { Priority } from "../value-object";

export type ID = number | string;

export type Reminder = "daily" | "three_days" | "seven_days";

export class Goal {
  constructor(
    public id: ID,
    public motivation: string,
    public reminder: Reminder,
    public tasks: Task[],
  ) {}
}

export type TaskStatus = "open" | "finished";

export class Task {
  constructor(
    public title: string,
    public description: string,
    public status: TaskStatus,
    public priority?: Priority,
    public id?: ID,
  ) {}
}
