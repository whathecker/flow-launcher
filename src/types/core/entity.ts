import { Priority } from "./value-object";

export type ID = number | string;

export type GoalStatus = "open" | "finished";
export type Reminder = "daily" | "three_days" | "seven_days";

export class Goal {
  constructor(
    public status: GoalStatus,
    public title: string,
    public motivation: string,
    public reminder: Reminder,
    public tasks: Task[],
    public _id?: ID,
  ) {}
}

export interface IAddGoalInput {
  title: string;
  motivation: string;
  reminder: string;
}

export interface IAddTaskInput {
  title: string;
  description: string;
  goal_id: string;
}

export interface IFetchTasksInput {
  goal_id: string;
}

export type TaskStatus = "open" | "finished";

export class Task {
  constructor(
    public _id: ID,
    public goal_id: ID,
    public title: string,
    public status: TaskStatus,
    public description?: string,
    public priority?: Priority,
  ) {}
}
