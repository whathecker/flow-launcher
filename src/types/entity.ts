import { Priority } from "./value-object";

export type ID = number | string;

export type GoalStatus = "open" | "finished";
export type Reminder = "daily" | "three_days" | "seven_days";

export class Goal {
  constructor(
    public status: GoalStatus,
    public title: string,
    public motivation: string,
    public reminder: Reminder,
    public tasks: Task[] | [],
    public _id?: ID,
  ) {}
}

export interface IAddGoalInput {
  title: string;
  motivation: string;
  reminder: string;
}

export type TaskStatus = "open" | "finished";

export class Task {
  constructor(
    public _id: ID,
    public goal_id: ID,
    public title: string,
    public description: string,
    public status: TaskStatus,
    public priority?: Priority,
  ) {}
}
