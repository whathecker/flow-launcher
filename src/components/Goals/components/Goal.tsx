import React from "react";
import { StyleSheet } from "react-native";
import { Touchable, Text } from "../../Themed";
import { Container, Typography, Shadow, Color } from "../../../styles";
import { Goal as GoalEntity } from "../../../types/core/entity";
import { navigationRef } from "../../../utils";

type GoalProps = {
  goal: GoalEntity;
  backgroundColor: string;
};

const Goal: React.FC<GoalProps> = ({ goal, backgroundColor }: GoalProps) => {
  const lightTextColor = Color.light.textOnColorForRead;
  const darkTextColor = Color.dark.textOnColorForRead;

  return (
    <Touchable
      lightColor={backgroundColor}
      darkColor={backgroundColor}
      style={styles.goalWrapper}
      onPress={() => {
        navigationRef.navigate("GoalDetail", {
          goal,
          goalColor: backgroundColor,
        });
      }}
    >
      <Text
        lightColor={lightTextColor}
        darkColor={darkTextColor}
        style={styles.goalTitleText}
      >
        {goal.title}
      </Text>
      <Text
        lightColor={lightTextColor}
        darkColor={darkTextColor}
        style={styles.goalMotiviationText}
      >
        {goal.motivation}
      </Text>
    </Touchable>
  );
};

const styles = StyleSheet.create({
  goalWrapper: {
    ...Container.centerAlignedVertical,
    alignItems: "flex-start",
    width: "90%",
    minHeight: 200,
    paddingLeft: 40,
    paddingRight: 60,
    paddingTop: 30,
    paddingBottom: 30,
    marginTop: 15,
    marginBottom: 15,
    marginLeft: "5%",
    marginRight: "5%",
    borderRadius: 10,
    ...Shadow.regularbackDrop,
  },
  goalTitleText: {
    ...Typography.h4,
    fontSize: 18,
    textAlign: "left",
    paddingBottom: 20,
  },
  goalMotiviationText: {
    ...Typography.p,
    fontSize: 16,
  },
});

export default Goal;
