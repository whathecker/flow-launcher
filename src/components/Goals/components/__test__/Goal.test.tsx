import * as React from "react";
import renderer from "react-test-renderer";
import Goal from "../Goal";
import { Goal as GoalEntity } from "../../../../types/core/entity";

describe("Test Goal component", () => {
  const dummyGoal: GoalEntity = {
    _id: "sjdiodfi103jsldicsd",
    title: "dummy goal -1",
    motivation: "this is my motivation",
    status: "open",
    reminder: "daily",
    tasks: [],
  };
  it("Should renders Goal correctly", () => {
    const tree = renderer
      .create(<Goal backgroundColor="#84C5E8" goal={dummyGoal} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
