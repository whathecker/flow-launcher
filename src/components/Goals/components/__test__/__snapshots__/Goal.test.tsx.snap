// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Goal component Should renders Goal correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "#84C5E8",
      "borderRadius": 10,
      "flexDirection": "column",
      "justifyContent": "center",
      "marginBottom": 15,
      "marginLeft": "5%",
      "marginRight": "5%",
      "marginTop": 15,
      "minHeight": 200,
      "opacity": 1,
      "paddingBottom": 30,
      "paddingLeft": 40,
      "paddingRight": 60,
      "paddingTop": 30,
      "shadowColor": "#171717",
      "shadowOffset": Object {
        "height": 5,
        "width": 3,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 3,
      "width": "90%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
        },
        Object {
          "fontFamily": "Lexend-SemiBold",
          "fontSize": 18,
          "paddingBottom": 20,
          "textAlign": "left",
        },
      ]
    }
  >
    dummy goal -1
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
        },
        Object {
          "fontFamily": "Lexend-Regular",
          "fontSize": 16,
        },
      ]
    }
  >
    this is my motivation
  </Text>
</View>
`;
